swap_meet/vendor.py

```diff
@@ -1,2 +1,77 @@
 class Vendor:
-    pass
+    def __init__(self,inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        # if inventory != None else inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False    
+
+    def get_by_category(self, category):
+        result = []
+        for each in self.inventory:
+            if each.category == category:
+                result.append(each)
+        return result
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in vendor.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            vendor.inventory.remove(their_item)
+            vendor.inventory.append(my_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) > 0 and len(vendor.inventory) > 0:
+            self.inventory.append(vendor.inventory[0])
+            vendor.inventory.append(self.inventory[0])
+            self.inventory.pop(0)
+            vendor.inventory.pop(0)
+            return True
+        else:
+            return False        
```

Comment:
This would have been a good spot to use`swap_items` as a helper method as well. Regardless, keeping `swap_first_item` and `swap_items` but have a time complexity of O(n) due to the `pop` and `remove` methods. We can improve the performance of `swap_first_item` by using multiple assignment:

```suggestion
    def swap_first_item(self, vendor):
        if len(self.inventory) == 0 or len(vendor.inventory) == 0:
            return False 
        self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
        return True
```

Here's a resource on time complexity  of Python's methods: https://wiki.python.org/moin/TimeComplexity

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp